lib/cc/cli/version_checker.rb

```diff
@@ -0,0 +1,69 @@
+require "uuid"
+
+module CC
+  module CLI
+    class VersionChecker
+      include CC::CLI::Output
+
+      VERSION_CHECK_TIMEOUT = 60 * 60 # 1 Hour in seconds
+      DEFAULT_VERSIONS_URL = "https://versions.codeclimate.com".freeze
+
+      def check
+        print_new_version_message if outdated?
+      rescue => error
+        CLI.debug(error)
+      end
+
+      private
+
+      def outdated?
+        api_response["outdated"] == true
+      end
+
+      def latest_version
+        api_response["latest"]
+      end
+
+      def print_new_version_message
+        warn "A new version (v#{latest_version}) is available"
+      end
+
+      def api_response
+        @api_response ||=
+          begin
```

Comment:
Sorry I missed this, my comment [here](https://github.com/codeclimate/codeclimate/pull/586#discussion_r104961767) applies here too. If you decide not to go that way then this comment can be ignored.